lib/matplotlib/cm.py

```diff
@@ -360,6 +360,11 @@ def to_rgba(self, x, alpha=None, bytes=False, norm=True):
                 else:
                     raise ValueError("Third dimension must be 3 or 4")
                 if xx.dtype.kind == 'f':
+                    # If any of R, G, B, or A is nan, set to 0
+                    if np.any(nans := np.isnan(x)):
+                        xx = xx.copy()
```

Comment:
We are now copying twice if the data is RGB _and_ has NaNs.  I'm not sure if that matters?